src/tests/functional/plugin/gpu/remote_blob_tests/gpu_remote_tensor_tests.cpp

```diff
@@ -720,6 +721,109 @@ TEST_P(OVRemoteTensor_TestsWithContext, smoke_canCreateManyTensorsOnSameMem) {
 INSTANTIATE_TEST_SUITE_P(smoke_RemoteTensor, OVRemoteTensor_TestsWithContext, ::testing::ValuesIn(ov_with_auto_batching),
                          OVRemoteTensor_TestsWithContext::getTestCaseName);
 
+TEST_F(OVRemoteTensor_Test, NV12_to_Grayscale) {
+#if defined(ANDROID)
+    GTEST_SKIP();
+#endif
+    const int height = 8;
+    const int width = 8;
+    const int feature = 1;
+
+    // ------------------------------------------------------
+    // Prepare input data
+    ov::Tensor fake_image = FuncTestUtils::create_and_fill_tensor(ov::element::i8, {1, feature, height, width}, 50, 0, 1);
+    ov::Tensor fake_image_regular = FuncTestUtils::create_and_fill_tensor(ov::element::f32, {1, feature, height, width});
+
+    auto image_ptr = static_cast<uint8_t*>(fake_image.data());
+    auto image_ptr_regular = static_cast<float*>(fake_image_regular.data());
+    // Apply NV12 (Surface) -> Gray conversion for regular blob
+    for (size_t i = 0; i < fake_image.get_size(); i++) {
+        auto val = static_cast<float>(image_ptr[i]) / 255;
+        val *= 296.82f;
+        val += -18.624f;
+        val = val < 0 ? 0 : val > 255 ? 255 : val;
+        image_ptr_regular[i] = val;
+    }
+
+    auto core = ov::Core();
+
+    // ------------------------------------------------------
+    // inference using remote tensor
+    auto fn_ptr_remote = ngraph::builder::subgraph::makeConvPoolRelu({1, feature, height, width});
```

Comment:
Will it work in batched case?

Reply:
Yes, I have added a batched func test and simple unit test, and also added pass which allows to change destination type of reorder through concat primitive in case of batched scenario